libvast/vast/system/node_control.hpp

```diff
@@ -63,7 +60,7 @@ get_node_components(caf::scoped_actor& self, const caf::actor& node) {
         result = detail::tuple_map<result_t>(std::move(components),
                                              detail::actor_cast_wrapper{});
       },
-      [&](caf::error& e) { //
+      [&](caf::error e) { //
```

Comment:
Why the copy when you dispatch the message?

Reply:
It's moved into the handler anyways. You either take by value or by const-reference, but taking by non-const-reference is just more prone to errors.

Reply:
It works slightly different, IIRC:

1. CAF dispatches the lambdas over a message
2. If you have a const-reference, you always a get a message to the contents
3. If you use a non-const-reference, you get mutable reference *iff* the ref-count is 1
4. If you use a plain value, you always get a copy at dispatch time.